src/app/public/src/modules/nav/nav.service.ts

```diff
@@ -1,160 +1,61 @@
+    if (this.isExternal(route)) {
+      this.windowRef.nativeWindow.location.href = route.path;
+      return;
     }
 
-    const rootPath = this.getActiveUrl().replace(/^\//, '').split('/')[0];
-    const appRoutes = this.clone(this.configService.runtime.routes);
-
-    let activeChildRoutes = appRoutes
-      .filter((route: any) => {
-        return (route.routePath.indexOf(rootPath) === 0);
-      })
-      .map((route: any) => {
-        return {
-          segments: route.routePath.split('/'),
-          path: route.routePath
-        };
-      });
-
-    let activeRoutes = [{
-      path: rootPath,
-      segments: [rootPath],
-      children: this.assignChildren(activeChildRoutes, rootPath)
-    }];
-
-    this.activeRoutes = this.formatRoutes(activeRoutes);
-
-    return this.clone(this.activeRoutes) as StacheNavLink[];
-  }
-
-  public getActiveUrl(): string {
-    return this.router.url.split('#')[0];
-  }
-
-  public clearActiveRoutes() {
-    this.activeRoutes = undefined;
-  }
-
-  private clone(thing: any): any {
-    return JSON.parse(JSON.stringify(thing));
-  }
+    if (route.fragment) {
+      const element = this.windowRef.nativeWindow.document.getElementById(route.fragment);
 
-  private assignChildren(routes: any[], parentPath: string): any[] {
-    const assignedRoutes: any[] = [];
-    const depth = parentPath.split('/').length + 1;
-
-    routes.forEach(route => {
-      const routeDepth = route.segments.length;
-      const isChildRoute = (depth === routeDepth && route.path.indexOf(parentPath) > -1);
-
-      if (isChildRoute) {
-        route.children = this.assignChildren(routes, route.path);
-        assignedRoutes.push(route);
+      if (element) {
+        this.scrollToElement(element, route.fragment);
+        return;
       }
-    });
-
-    return assignedRoutes;
-  }
-
-  private formatRoutes(routes: any[]): StacheNavLink[] {
-    let formatted = routes.map(route => {
-      let pathMetadata = this.getRouteMetadata(route);
 
-      let formattedRoute = Object.assign({},
-        {
-          path: route.path,
-          children: this.formatRoutes(route.children),
-          name: this.getNameFromPath(route.segments[route.segments.length - 1])
-        },
-        pathMetadata);
-
-      return formattedRoute as any;
-    });
-
-    return this.sortRoutes(formatted) as StacheNavLink[];
-  }
-
-  private getRouteMetadata(route: any): any {
-    const allMetadata = this.routeMetadataService.metadata;
-
-    if (allMetadata) {
-      let foundRoute = allMetadata.filter((metaRoute: any) => {
-        return metaRoute.path === route.path;
-      })[0];
-
-      if (foundRoute) {
-        return foundRoute;
+      // The current page is the path intended, but no element with the fragment exists, scroll to
+      // the top of the page and remove the hash from the url.
+      if (route.path === currentPath) {
+        this.windowRef.nativeWindow.scroll(0, 0);
+        this.windowRef.nativeWindow.location.hash = '';
```

Comment:
6 of 1 half dozen of the other.   It could be helpful if there is actually a broken hash someone tried to link for some reason to see where it went wrong.   Or it could just be distracting to have a hash that doesn't do anything but brings you to the top of the page.